protobuf-provider/src/main/java/io/confluent/kafka/schemaregistry/protobuf/ProtobufSchema.java

```diff
@@ -0,0 +1,707 @@
+            .toString()).orElse(null);
+        oneofBuilder.addField(
+            field.getType(),
+            field.getName(),
+            field.getTag(),
+            defaultVal,
+            jsonName
+        );
+        added.add(field.getName());
+      }
+    }
+    for (FieldElement field : messageElem.getFields()) {
+      if (added.contains(field.getName())) {
+        continue;
+      }
+      String label = field.getLabel() != null ? field.getLabel().toString().toLowerCase() : null;
+      String fieldType = field.getType();
+      String defaultVal = field.getDefaultValue();
+      String jsonName = findOption("json_name", field.getOptions()).map(o -> o.getValue()
+          .toString()).orElse(null);
+      Boolean isPacked = findOption(
+          "packed",
+          field.getOptions()
+      ).map(o -> Boolean.valueOf(o.getValue().toString())).orElse(null);
+      ProtoType protoType = ProtoType.get(fieldType);
+      // Map fields are only permitted in messages
+      if (protoType.isMap()) {
+        label = "repeated";
+        fieldType = toMapEntry(field.getName());
+        MessageDefinition.Builder mapMessage = MessageDefinition.newBuilder(fieldType);
+        mapMessage.setMapEntry(true);
+        mapMessage.addField(null, protoType.keyType().simpleName(), KEY_FIELD, 1, null);
+        mapMessage.addField(null, protoType.valueType().simpleName(), VALUE_FIELD, 2, null);
+        schema.addMessageDefinition(mapMessage.build());
+      }
+      message.addField(
+          label,
+          fieldType,
+          field.getName(),
+          field.getTag(),
+          defaultVal,
+          jsonName,
+          isPacked
+      );
+    }
+    for (TypeElement type : messageElem.getNestedTypes()) {
+      if (type instanceof MessageElement) {
+        message.addMessageDefinition(toDynamicMessage(schema, (MessageElement) type));
+      } else if (type instanceof EnumElement) {
+        message.addEnumDefinition(toDynamicEnum((EnumElement) type));
+      }
+    }
+    for (ReservedElement reserved : messageElem.getReserveds()) {
+      for (Object elem : reserved.getValues()) {
+        if (elem instanceof String) {
+          message.addReservedName((String) elem);
+        } else if (elem instanceof Integer) {
+          int tag = (Integer) elem;
+          message.addReservedRange(tag, tag);
+        } else if (elem instanceof Range) {
+          Range<Integer> range = (Range<Integer>) elem;
+          message.addReservedRange(range.lowerEndpoint(), range.upperEndpoint());
+        } else {
+          throw new IllegalStateException("Unsupported reserved type: " + elem.getClass()
+              .getName());
+        }
+      }
+    }
+    return message.build();
+  }
+
+  private static Optional<OptionElement> findOption(String name, List<OptionElement> options) {
+    return options.stream().filter(o -> o.getName().equals(name)).findFirst();
+  }
```

Comment:
Question: can this ever return null ?

Reply:
No, it will always be a non-null Optional